linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_faillock_deny/oval/shared.xml

```diff
@@ -1,222 +1,290 @@
 <def-group>
-  <definition class="compliance" id="accounts_passwords_pam_faillock_deny" version="4">
-    {{{ oval_metadata("The number of allowed failed logins should be set correctly.") }}}
-    <criteria operator="AND" comment="Checks common to both scenarios">
-      <criterion test_ref="test_accounts_passwords_pam_faillock_preauth_silent_system-auth"
-      comment="pam_faillock.so preauth silent set in system-auth" />
-      <criterion test_ref="test_accounts_passwords_pam_faillock_account_phase_system-auth"
-      comment="pam_faillock.so set in account phase of system-auth" />
-      <criterion test_ref="test_accounts_passwords_pam_faillock_preauth_silent_password-auth"
-      comment="pam_faillock.so preauth silent set in password-auth" />
-      <criterion test_ref="test_accounts_passwords_pam_faillock_account_phase_password-auth"
-      comment="pam_faillock.so set in account phase of password-auth" />
-      <criteria operator="AND">
-        <criteria operator="OR" comment="system-auth">
-          <criterion test_ref="test_accounts_passwords_pam_faillock_numeric_default_check_system-auth"
-            comment="Perform check if pam_faillock authfail follows pam_unix even with lines skipped" />
-          <criterion test_ref="test_accounts_passwords_pam_faillock_authfail_deny_system-auth"
-            comment="Perform check if pam_faillock authfail follows pam_unix with either sufficient or default=ignore" />
+  <definition class="compliance" id="{{{ rule_id }}}" version="5">
+    {{{ oval_metadata("Lockout account after failed login attempts") }}}
+    <criteria operator="AND" comment="Check the proper configuration of pam_faillock.so">
+      <criteria operator="AND" comment="Check if pam_faillock.so is properly enabled">
+        <!-- pam_unix.so is a control module present in all realistic scenarios and also used
+             as reference for the correct position of pam_faillock.so in auth section. If the
+             system is properly configured, it must appear only once in auth section. -->
+        <criteria operator="AND" comment="Count occurrencies of pam_unix.so in system-auth">
```

Comment:
Nitpick: the comment is a bit misleading because the 2 criteria are one about system-auth but one about password-auth.
```suggestion
        <criteria operator="AND" comment="Count occurrencies of pam_unix.so in system-auth and password-auth">
```